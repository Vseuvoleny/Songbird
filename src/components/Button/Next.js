import React from "react";
import classes from "./Next.module.css";

const Next = props => {
  const cls = [classes.Next];
  props.disabled ? cls.push(classes.inactive) : cls.push(classes.active);

  return (
    <div className={classes.btn}>
      <button
        disabled={props.disabled}
        className={cls.join(" ")}
        onClick={props.NextClickHandler}
      >
        Next Level
      </button>
    </div>
  );
};

export default Next;
