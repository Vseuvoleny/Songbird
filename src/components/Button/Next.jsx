import React from "react";
import { connect } from "react-redux";
import classes from "./Next.module.css";

const Next = ({ isAnswerRight, nextQuestion }) => {
  const cls = [classes.Next];
  isAnswerRight ? cls.push(classes.active) : cls.push(classes.inactive);

  return (
    <div className={classes.btn}>
      <button
        disabled={!isAnswerRight}
        className={cls.join(" ")}
        onClick={nextQuestion}
      >
        Next Level
      </button>
    </div>
  );
};

function mapStateToProps(state) {
  return {
    isAnswerRight: state.isAnswerRight
  };
}

function mapDispatchToProps(dispatch) {
  return {
    nextQuestion: () => dispatch({ type: "NEXT_QUESTION" })
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Next);
