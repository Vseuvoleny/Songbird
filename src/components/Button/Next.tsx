import React from "react";
import { connect } from "react-redux";
import "./styles.scss";

type NextButton = {
  isAnswerRight: boolean;
  nextQuestion: () => {};
};

const Next: React.FC<NextButton> = ({ isAnswerRight, nextQuestion }) => {
  return (
    <div className="button_next_container">
      <button
        disabled={!isAnswerRight}
        className="button"
        onClick={nextQuestion}
      >
        Next Level
      </button>
    </div>
  );
};

export default connect(
  (state: any) => ({ isAnswerRight: state.isAnswerRight }),
  (dispatch) => ({ nextQuestion: () => dispatch({ type: "NEXT_QUESTION" }) })
)(Next);
