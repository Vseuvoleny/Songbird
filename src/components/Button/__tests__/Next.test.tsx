import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import Next from "../Next";

describe("Next", () => {
  test("should render Next button component", () => {
    const handleClick = jest.fn();
    render(<Next isAnswerRight={true} nextQuestion={handleClick} />);
    expect(screen.getByText(/Next Level/i)).toBeInTheDocument();
    expect(screen.getByText(/Next Level/i)).not.toBeDisabled();
  });

  test("should clicked", () => {
    const handleClick = jest.fn();
    const { container } = render(
      <Next isAnswerRight={true} nextQuestion={handleClick} />
    );
    expect(container).toMatchSnapshot();
    fireEvent.click(screen.getByText(/Next Level/i));
    expect(handleClick).toHaveBeenCalled();
  });
});
