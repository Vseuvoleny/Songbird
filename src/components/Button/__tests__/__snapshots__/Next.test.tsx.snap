// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Next should clicked 1`] = `
<div>
  <div
    class="button_next_container"
  >
    <button
      class="button"
    >
      Next Level
    </button>
  </div>
</div>
`;
